Flaky failures in `test_concurrent_lazy_init`

`make test` sometimes fails at HEAD on my laptop:

```
bryan@bryan-macbook-pro:~/src/minject$ git status
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean

bryan@bryan-macbook-pro:~/src/minject$ git rev-parse HEAD
1439db9c886275f02b971573192c515e0c39881a

bryan@bryan-macbook-pro:~/src/minject$ make test
.venv/bin/python3 -m hatch run python docs/examples/philosophy.py
.venv/bin/python3 -m hatch run python -m unittest discover -s tests
philosophy example tests passed!
.....F..................
======================================================================
FAIL: test_concurrent_lazy_init (test_registry.RegistryTestCase.test_concurrent_lazy_init)
Test lazy initialization of singletons in a concurrent environment always returns the same object
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/bryan/src/minject/tests/test_registry.py", line 481, in test_concurrent_lazy_init
    assert all(count == query_per_class for count in counter.values())
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError

----------------------------------------------------------------------
Ran 24 tests in 0.034s

FAILED (failures=1)
make: *** [test-unit] Error 1
```

(CC @mmchenry-duolingo @biniona)

--- a/tests/test_registry.py
+++ b/tests/test_registry.py
@@ -1,7 +1,6 @@
 import unittest
 from collections import Counter
 from concurrent.futures import ThreadPoolExecutor, as_completed
-from functools import lru_cache
 from random import shuffle
 from typing import Sequence
 
@@ -464,12 +463,10 @@ class RegistryTestCase(unittest.TestCase):
         query_per_class = 2
         num_classes = num_queries // query_per_class
 
-        @lru_cache(maxsize=None)
-        def new_type(i):
-            return type(f"NewType{i}", (), {})
+        types = [type(f"NewType{i}", (), {}) for i in range(num_classes)]
 
         def lazy_load_object(i):
-            return self.registry[new_type(i % num_classes)]
+            return self.registry[types[i % num_classes]]
 
         with ThreadPoolExecutor(max_workers=query_per_class) as executor:
             futures = [executor.submit(lazy_load_object, i) for i in range(num_queries)]
@@ -477,8 +474,8 @@ class RegistryTestCase(unittest.TestCase):
 
         # group results by object id, and assert that each type is only instantiated once If each type is
         # instantiated only once but accessed N times, we would see N objects for each type in the counter.
-        counter = Counter(map(id, results))
-        assert all(count == query_per_class for count in counter.values())
+        for count in Counter(map(id, results)).values():
+            self.assertEqual(count, query_per_class)
 
 
 # "Test"/check type hints.  These are not meant to be run by the unit test runner, but instead to

